xcode/DirectBindingsApp/DirectBindingsApp/app-batteries/BlockchainObserver.swift

```diff
@@ -478,7 +478,8 @@ class BlockchainObserver {
 		return result
 	}
 
-    public func isMonitoring() async -> Bool {
+    public func isMonitoring() async throws -> Bool {
+        try await preloadMonitor()
```

Comment:
sorry, that makes it worse. You don't want a check (basically a getter) to change state